doc/source/user_guide/text.rst

```diff
@@ -29,7 +30,15 @@ for many reasons:
 3. When reading code, the contents of an ``object`` dtype array is less clear
    than ``'string'``.
 
-Currently, the performance of ``object`` dtype arrays of strings and
+Starting from pandas 3.0, the default dtype for string data is ``str``. Previously, ``object`` dtype was the
+default for string data. You may still encounter ``object`` dtype for legacy datasets or while working with earlier
+pandas versions. Use ``.astype("str")`` to explicitly convert these to ``str`` dtype or specify ``dtype="str"``
+when creating new data structures.
+
+Use the nullable :class:`StringDtype` (``"string"``) when handling NA values in your string data. It offers
+additional flexibility for missing values while maintaining compatibility with pandas' nullable types.
```

Comment:
``str`` also handles NA values, the difference between `np.NaN` vs `pd.NA`. I think it'd be good to clarify that here.